aten/src/ATen/core/PhiloxRNGEngine.h

```diff
@@ -192,12 +190,55 @@ class philox_engine {
     ret[3] = lo0;
     return ret;
   }
+
+  C10_HOST_DEVICE constexpr float normalize(uint32_t value) {
+    int32_t x = (int32_t)value;
+    float scale = 4.6566127342e-10;
+    if (x < 0) {
+      x = -x - 1;
+    }
+    return 1.0f - static_cast<float>(x * scale);
```

Comment:
Why are we generating 0-exclusive, 1-inclusive here (`(0,1]`), don't people usually expect `[0,1)`?

Reply:
I think It has to exclude 0 if it goes to log below?

Reply:
So you dont have to search, I am talking about `randn`, where we normalize the inputs into `normalize_pair_uniform`, and call `log` on `u1`

Reply:
Further thinking - If we want this to be a true normalize for general purpose, it should be `[0,1)` and then in `normalize_pair_uniform` we can do `1-r`.